### PR TITLE
[Feature] Add Contrib/Non-free Tag

### PR DESCRIPTION
**Pull request by asddsaz**
**Description**: Add contrib tag to software that recommends or depends on non-free software.
**What is contrib**? Contrib is a term created by the [Debian Foundation](https://www.wikipedia.org/wiki/Debian) to discern from software that is free software but may depend on non-free software.
**But, isn't everything listed on privacytools.io only free software?** Nope, especially not OSes of which many will auto-install non-free firmware like [QubesOS](https://www.reddit.com/r/fsf/comments/8ue7fi/thoughts_on_qubes_os/).

### CODE REVIEW

**Comment by Shifterovich** [comment]
Please add the labels the same way Brave's experimental label is added.

**Comment by Shifterovich** [comment]
Note that you can't use `"` or `,` in the labels attribute. I will write a wiki page about this sometime.

For commas you can use `&comma;`.

**Comment by Mikaela** [comment]
Should something be done to apps that are free and use free servers, but where server is not configurable, for example Signal?

**Comment by Shifterovich** [comment]
With Wire we warn about plaintext metadata. If Signal's open source frontend encrypts everything, there's nothing to warn about.

**Comment by asddsaz** [comment]
> Please add the labels the same way Brave's experimental label is added.

@Shifterovich Done!

File: index.html [hunk 1]
```diff
@@ -1895,13 +1896,15 @@ <h1 id="live_os" class="anchor"><a href="#live_os"><i class="fas fa-link anchor-
 
     {% include card.html color="success"
     title="Tails"
+    labels="warning:contrib:This software may depend on or recommend non-free software."
     image="assets/img/tools/Tails.png"
     url="https://tails.boum.org/"
     description='<span class="badge badge-warning" data-toggle="tooltip" title="This software may depend on or recommend non-free software.">contrib <i class="far fa-question-circle"></i></span> Tails is a live operating system, that starts on almost any computer from a DVD, USB stick, or SD card. It aims at preserving privacy and anonymity, and helps to: Use the Internet anonymously and circumvent censorship; Internet connections go through the Tor network; leave no trace on the computer; use state-of-the-art cryptographic tools to encrypt files, emails and instant messaging.'
```
Comment (Shifterovich) [inline index.html]: You left the HTML in the description here.

File: index.html [hunk 2]
```diff
@@ -1910,6 +1913,7 @@ <h1 id="live_os" class="anchor"><a href="#live_os"><i class="fas fa-link anchor-
 
     {% include card.html color="warning"
     title="Puppy Linux"
+    labels="warning:contrib:This software may depend on or recommend non-free software."
     image="assets/img/tools/Puppy-Linux.png"
     url="http://puppylinux.org/"
     description='<span class="badge badge-warning" data-toggle="tooltip" title="This software may depend on or recommend non-free software.">contrib <i class="far fa-question-circle"></i></span> Puppy Linux operating system is a lightweight Linux distribution that focuses on ease of use and minimal memory footprint. The entire system can be run from RAM with current versions generally taking up about 210 MB, allowing the boot medium to be removed after the operating system has started.'
```
Comment (Shifterovich) [inline index.html]: You left the HTML in the description here.

File: index.html [hunk 3]
```diff
@@ -1930,6 +1934,7 @@ <h1 id="mobile_os" class="anchor"><a href="#mobile_os"><i class="fas fa-link anc
 
     {% include card.html color="success"
     title="LineageOS"
+    labels="warning:contrib:This software may depend on or recommend non-free software."
     image="assets/img/tools/LineageOS.png"
     url="https://www.lineageos.org/"
     description='<span class="badge badge-warning" data-toggle="tooltip" title="This software may depend on or recommend non-free software.">contrib <i class="far fa-question-circle"></i></span> LineageOS is a free and open-source operating system for smartphones and tablets, based on the official releases of Android by Google. It is the continuation of the CyanogenMod project.'
```
Comment (Shifterovich) [inline index.html]: You left the HTML in the description here.

File: index.html [hunk 4]
```diff
@@ -1970,13 +1975,15 @@ <h3>Manage your apps' permissions with XPrivacyLua</h3>
...
     %}
 
     {% include card.html color="primary"
     title="pfSense"
+    labels="warning:contrib:This software may depend on or recommend non-free software."
     image="assets/img/tools/pfSense.png"
     url="https://pfsense.org/"
     description='<span class="badge badge-warning" data-toggle="tooltip" title="This software may depend on or recommend non-free software.">contrib <i class="far fa-question-circle"></i></span> pfSense is an open source firewall/router computer software distribution based on FreeBSD. It is installed on a computer to make a dedicated firewall/router for a network and is noted for its reliability and offering features often only found in expensive commercial firewalls. pfSense is commonly deployed as a perimeter firewall, router, wireless access point, DHCP server, DNS server, and as a VPN endpoint.'
```
Comment (Shifterovich) [inline index.html]: You left the HTML in the description here.

File: index.html [hunk 5]
```diff
@@ -1970,13 +1975,15 @@ <h3>Manage your apps' permissions with XPrivacyLua</h3>
 
     {% include card.html color="success"
     title="OpenWrt"
+    labels="warning:contrib:This software may depend on or recommend non-free software."
     image="assets/img/tools/OpenWrt.png"
     url="https://openwrt.org/"
     description='<span class="badge badge-warning" data-toggle="tooltip" title="This software may depend on or recommend non-free software.">contrib <i class="far fa-question-circle"></i></span> OpenWrt is an operating system (in particular, an embedded operating system) based on the Linux kernel, primarily used on embedded devices to route network traffic. The main components are the Linux kernel, util-linux, uClibc and BusyBox. All components have been optimized for size, to be small enough for fitting into the limited storage and memory available in home routers.'
```
Comment (Shifterovich) [inline index.html]: You left the HTML in the description here.

**Comment by asddsaz** [comment]
@Shifterovich Would you mind plz explaining the requested changes. I do not understand. :)

**Comment by Shifterovich** [comment]
You added the `labels=...` but kept the `<span class="badge badge-warning" data-toggle="tooltip" title="This  ...` in the `description`.

**Comment by asddsaz** [comment]
@Shifterovich Hmm, try re-pulling. I only see it for LineageOS. I made sure to fix that. :)
Thank you so much for your help!

**Comment by asddsaz** [comment]
@Shifterovich So, one example would be that Ubuntu "recommends" non-free software like Steam. 

The main issue with the items I want to currently add this to is firmware but if an OS/App recommends a user install non-free software you could use the same tag.
Preferably, it would have a more specific warning like:

> This software may recommend non-free software such as apps or addons.

It is meant to warn the user that although the app is free software what it may ask to install is non.

**Comment by Shifterovich** [comment]
In what way does Ubuntu recommend Steam? (If it does, I haven't noticed.)

**Comment by asddsaz** [comment]
> In what way does Ubuntu recommend Steam? (If it does, I haven't noticed.)

During installation they show many apps available for Ubuntu. They also have many non-free apps in the App Store.

The [FSF](https://www.fsf.org)/[GNU](http://www.gnu.org) has a good article on this:

> Ubuntu provides specific repositories of nonfree software, and Canonical expressly promotes and recommends nonfree software under the Ubuntu name in some of their distribution channels. Ubuntu offers the option to install only free packages, which means it also offers the option to install nonfree packages too. In addition, the version of Linux, the kernel, included in Ubuntu contains firmware blobs.
> 
> The “Ubuntu Software Center” lists proprietary programs and free programs jumbled together. It is hard to tell which ones are free since proprietary programs for download at no charge are labelled “free”.

Source: http://www.gnu.org/distros/common-distros.html

File: index.html [hunk 6]
```diff
@@ -1851,10 +1851,11 @@ <h1 id="os" class="anchor"><a href="#os"><i class="fas fa-link anchor-icon"></i>
 
     {% include card.html color="success"
     title="Qubes OS"
+    labels="warning:contrib:This software may depend on or recommend non-free software."
     image="assets/img/tools/Qubes-OS.png"
     url="https://www.qubes-os.org/"
     tor="http://qubesos4rrrrz6n4.onion"
-    description="Qubes is an open-source operating system designed to provide strong security for desktop computing. Qubes is based on Xen, the X Window System, and Linux, and can run most Linux applications and utilize most of the Linux drivers."
+    description='Qubes is an open-source operating system designed to provide strong security for desktop computing. Qubes is based on Xen, the X Window System, and Linux, and can run most Linux applications and utilize most of the Linux drivers.'
```
[the remark this answers is not in the document]
Comment (Shifterovich) [reply index.html]: Since is no painful escaping is solved by this, this should be reverted to be consistent with the include's other properties.

[the remark this answers is not in the document]
Comment (asddsaz) [reply index.html]: @Vincevrp @Shifterovich Done! (https://github.com/privacytoolsIO/privacytools.io/pull/687/commits/fa6f752fe4e57f839eef9f3454e088a9a169e970)

If I remember correctly, the reasons I did this was because ' " ' was in the description. Therefore I had to switch on order for it to compile properly. But, this should no longer be a problem.

**Comment by asddsaz** [comment]
> Could you please add some screenshots?

Sure, I'll edit this post when I've downloaded and complied it all....

**Comment by asddsaz** [comment]
selection_019

**Comment by BurungHantu1605** [comment]
hi guys, whats the status here?

**Comment by asddsaz** [comment]
@Vincevrp

**Comment by BurungHantu1605** [comment]
I don't think we need this feature at the moment.

**Comment by Mikaela** [comment]
There is still a merge conflict.

I think especially the privacy concious users of https://github.com/privacytoolsIO/privacytools.io/issues/880 might appreciate this.

**Comment by asddsaz** [comment]
> This PR seems to have became outdated/merge conflicting due to the website update. I would also like to see how this looks in practice and I think Netlify integration would kick in and bring a preview when the PR is updated.

This should have been fixed in last [commit/PR](https://github.com/privacytoolsIO/privacytools.io/pull/687/commits/896bfb6261275086ac5c912551425abe9e24f46a).

File: _includes/sections/live-operating-systems.html [hunk 7]
```diff
@@ -4,13 +4,15 @@ <h1 id="live_os" class="anchor"><a href="#live_os"><i class="fas fa-link anchor-
 
   {% include card.html color="success"
   title="Tails"
+  labels="warning:contrib:This software may depend on or recommend non-free software."
   image="/assets/img/tools/Tails.png"
   url="https://tails.boum.org/"
-  description="Tails is a live operating system, that starts on almost any computer from a DVD, USB stick, or SD card. It aims at preserving privacy and anonymity, and helps to: Use the Internet anonymously and circumvent censorship; Internet connections go through the Tor network; leave no trace on the computer; use state-of-the-art cryptographic tools to encrypt files, emails and instant messaging."
+  description='Tails is a live operating system, that starts on almost any computer from a DVD, USB stick, or SD card. It aims at preserving privacy and anonymity, and helps to: Use the Internet anonymously and circumvent censorship; Internet connections go through the Tor network; leave no trace on the computer; use state-of-the-art cryptographic tools to encrypt files, emails and instant messaging.'
```
Comment (Mikaela) [inline _includes/sections/live-operating-systems.html]: Why the change of " to ' ?

**Comment by blacklight447-ptio** [comment]
Seems fine to me:+1:

**Comment by Mikaela** [comment]
There are three OKs, but @blacklight447-ptio didn't catch the review approval button being in top right of "files changed".

**Comment by JonahAragon** [comment]
I'm more just not 100% sure if noting `contrib` on not completely free tools is in-scope for this project. It might make people more wary of using certain tools when in reality if people used *any* of the tools we recommend it would be a significant net benefit to privacy overall.

But I also don't have any problem with this being added.

**Comment by Shifterovich** [comment]
A downside is that it will pollute the source code and the site with a ton of `contrib` labels.

**Comment by asddsaz** [comment]
> A downside is that it will pollute the source code and the site with a ton of `contrib` labels.
> 
> > I'm more just not 100% sure if noting contrib on not completely free tools is in-scope for this project. [...] But I also don't have any problem with this being added.
> 
> We should only add things if there's a benefit to adding them.

@Shifterovich I'm not sure I understand, this is already part of my PR.
I didn't add it to all the non-free search engines and hosting sites. I also didn't add it to services like Signal or search engines or even the tons of VPNs with proprietary back-ends.

I any body would like discussing adding the label to these, I'de be open to it.
But adding it to OSes is an important role. Many sites and resources don't list it.

Personally, I don't want non-free firmware installed. I don't use  my laptops built-in Bluetooth because it requires drivers I cannot audit or change. 
I also rely on my OSes repos to be transparent and remove or warn me of software that isn't free.

People just don't understand the flaws with OSes like Ubuntu and Qubes. 
Warning people allows them to make their own choice. It adds what is very similar to a warning label.

**Comment by Shifterovich** [comment]
I didn't look at the diff and automatically assumed the label was added to a ton of cards, sorry.

My opinion is the same as @JonahAragon's.

**Comment by asddsaz** [comment]
There appears to have been couple of contrib OSes I must've forgotten to add the label to.
No need to worry I've already added them to the PR. But, wanted to inform you on why the extra commits.

**Comment by asddsaz** [comment]
Aye, I was just curious if any of the past reviews have changed.
If they have, is there any changes that you would like undone or reverted or changed?

There hasn't been a discussion here is a while. :)

 @JonahAragon @Shifterovich @Mikaela

File: index.html [hunk 8]
```diff
@@ -7,4 +7,4 @@
 {% include sections/resources.html %}
 {% include sections/quotes.html %}
 {% include sections/privacy-resources.html %}
-{% include sections/participate.html %}
+{% include sections/participate.html %}
```
Comment (JonahAragon) [inline index.html]:
```suggestion
{% include sections/participate.html %}

```

This is not good style.

File: _includes/sections/live-operating-systems.html (hunk 7 above)
Comment (JonahAragon) [reply _includes/sections/live-operating-systems.html]: I'd also like to know this? Seems inconsistent.

```suggestion
  description="Tails is a live operating system, that starts on almost any computer from a DVD, USB stick, or SD card. It aims at preserving privacy and anonymity, and helps to: Use the Internet anonymously and circumvent censorship; Internet connections go through the Tor network; leave no trace on the computer; use state-of-the-art cryptographic tools to encrypt files, emails and instant messaging."
```

File: index.html (hunk 8 above)
Comment (asddsaz) [reply index.html]: It does seem to make any difference to the visual side.

I'll commit suggestion.

File: _includes/sections/live-operating-systems.html (hunk 7 above)
Comment (asddsaz) [reply _includes/sections/live-operating-systems.html]: Should be fixed in later commit. :)

Thanks!

**Comment by JonahAragon** [comment]
Okay well, I suppose I'll go ahead and merge this 🤞
